src/utils/aggregation.jl

```diff
@@ -1,74 +1,147 @@
+struct AggregatedResults <: Results
+    variables::Dict{Symbol, DataFrames.DataFrame}
+    total_cost::Dict{Symbol, Any}
+    optimizer_log::Dict{Symbol, Any}
+    time_stamp::DataFrames.DataFrame
+    duals::Dict{Symbol, Any}
+end
+
+get_res_variables(result::AggregatedResults) = result.variables 
+get_cost(result::AggregatedResults) = result.total_cost
+get_log(result::AggregatedResults) = result.optimizer_log
+get_time_stamp(result::AggregatedResults) = result.time_stamp
+get_duals(result::AggregatedResults) = result.duals
+get_variables(result::AggregatedResults) = result.dual_variables
+
 # This function returns the length of time_stamp for each step
 # that is unique to that step and not overlapping with the next time step.
-
 function _count_time_overlap(stage::String,
     step::Array,
     variable::Array,
     references::Dict{Any,Any})
-
-
-date_df = references[stage][variable[1]]
-step_df = DataFrames.DataFrame(Date = Dates.DateTime[],
-           Step = String[],
-           File_Path = String[])
-
-for n in 1:length(step)
-step_df = vcat(step_df,date_df[date_df.Step .== step[n], :])
-
-end
-ref = DataFrames.DataFrame()
-
-for (ix,time) in enumerate(step_df.Date)
-
-try
-file_path =  step_df[ix, :File_Path]
-time_file_path = joinpath(dirname(file_path), "time_stamp.feather")
-temp_time_stamp = DataFrames.DataFrame(Feather.read("$time_file_path"))
-t = size(temp_time_stamp, 1)
-append!(ref,temp_time_stamp[(1:t-1),:])
-
-catch
-@warn "The given date_range is outside the results time stamp."
-end
-end
-if size(unique(ref),2) == size(ref,2)
-extra_time_length = 0
-else
-extra_time_length = size(unique(ref),1)./(length(step)+1)
-end
-return extra_time_length
+    date_df = references[stage][variable[1]]
+    step_df = DataFrames.DataFrame(Date = Dates.DateTime[], 
+            Step = String[], 
+            File_Path = String[])
+    for n in 1:length(step)
+        step_df = vcat(step_df,date_df[date_df.Step .== step[n], :])
+    end
+    ref = DataFrames.DataFrame()
+    for (ix,time) in enumerate(step_df.Date)
+        try
+            file_path =  step_df[ix, :File_Path]
+            time_file_path = joinpath(dirname(file_path), "time_stamp.feather")
+            time_stamp = DataFrames.DataFrame(Feather.read("$time_file_path"))
+            shorten_time_stamp!(time_stamp)
+            append!(ref,time_stamp)
+        catch
```

Comment:
You could explicitly detect the bad date range without relying on the exception.  In this case if someone makes any coding error that causes an exception, it will manifest as this bad date range.